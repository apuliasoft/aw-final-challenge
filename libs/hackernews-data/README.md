# hackernews-data

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test hackernews-data` to execute the unit tests.
