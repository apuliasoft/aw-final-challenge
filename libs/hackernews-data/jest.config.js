module.exports = {
  name: 'hackernews-data',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/libs/hackernews-data',
  snapshotSerializers: [
    'jest-preset-angular/AngularSnapshotSerializer.js',
    'jest-preset-angular/HTMLCommentSerializer.js'
  ]
};
