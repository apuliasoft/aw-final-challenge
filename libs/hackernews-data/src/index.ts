export * from './lib/hackernews-data.module';
export * from './lib/types';
export * from './lib/state/news/news.facade';
