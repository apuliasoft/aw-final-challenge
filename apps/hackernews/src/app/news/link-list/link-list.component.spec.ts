import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { NO_ERRORS_SCHEMA, Pipe, PipeTransform } from '@angular/core';
import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { of } from 'rxjs';

import { NewsFacade, Link } from '@hn/hackernews-data';

import { LinkListComponent } from './link-list.component';

const links: Link[] = [
  {
    id: 1,
    description: 'Best search engine, ever',
    url: 'https://www.google.com',
    points: 0,
    pubDate: new Date()
  }
];

const newsFacadeMock = {
  links$: of(links),
  loadLinks: jest.fn()
};

@Pipe({
  name: 'simpleUrl'
})
export class SimpleUrlPipeMock implements PipeTransform {
  transform(value: string): string {
    return 'www.google.com';
  }
}

@Pipe({
  name: 'timeAgo'
})
export class TimeAgoPipeMock implements PipeTransform {
  transform(value: Date, withoutSuffix?: boolean): string {
    return 'a few seconds';
  }
}

describe('LinkListComponent', () => {
  let sut: ComponentFixture<LinkListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [NO_ERRORS_SCHEMA],
      imports: [NoopAnimationsModule, RouterTestingModule],
      declarations: [LinkListComponent, SimpleUrlPipeMock, TimeAgoPipeMock],
      providers: [
        {
          provide: NewsFacade,
          useValue: newsFacadeMock
        }
      ]
    }).compileComponents();

    sut = TestBed.createComponent(LinkListComponent);
    sut.detectChanges();
  }));

  test('should load links list', () => {
    expect(newsFacadeMock.loadLinks).toBeCalledWith(1);
  });

  test('should render links list', () => {
    expect(sut).toMatchSnapshot();
  });
});
