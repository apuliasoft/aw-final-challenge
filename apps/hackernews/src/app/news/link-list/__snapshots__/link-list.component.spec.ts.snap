// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkListComponent should render links list 1`] = `
<hn-link-list
  links$={[Function Observable]}
  newsFacade={[Function Object]}
  page={[Function Number]}
  route={[Function ActivatedRoute]}
  scroll$={[Function Observable]}
>
  <div
    class="ng-tns-c1-0 ng-trigger ng-trigger-fadeIn ng-star-inserted"
    style=""
  >
    
    <div
      class="special-dark-theme ng-tns-c1-0 ng-star-inserted"
      style=""
    >
      <mat-card
        class="card"
      >
        <mat-card-header
          class="ng-tns-c1-0"
        >
          <mat-card-title
            class="ng-tns-c1-0"
          >
             Best search engine, ever 
          </mat-card-title>
          <mat-card-subtitle
            class="ng-tns-c1-0"
          >
             www.google.com (a few seconds) 
          </mat-card-subtitle>
        </mat-card-header>
        <mat-card-actions
          class="ng-tns-c1-0"
        >
          <button
            class="ng-tns-c1-0"
            color="primary"
            mat-raised-button=""
            matbadgecolor="accent"
            matbadgeposition="before"
          >
             UPVOTE 
          </button>
          <a
            class="ng-tns-c1-0"
            color="accent"
            href="https://www.google.com"
            mat-raised-button=""
            target="new"
          >
             GO TO LINK 
          </a>
        </mat-card-actions>
      </mat-card>
    </div>
  </div><a
    class="more ng-trigger ng-trigger-scroll"
    color="accent"
    href="/new?p=2"
    mat-fab=""
    ng-reflect-query-params="[object Object]"
    ng-reflect-router-link="/new"
    style=""
  >
    More
  </a>
</hn-link-list>
`;
